File: cse241-object-oriented-programming/studies/java/inheritance/employee/SalariedEmployee.java
package employee;

public class SalariedEmployee extends Employee {
    private double _weeklySalary;

    public SalariedEmployee (String fname, String lname, String ssn, double salary) {
        super(fname, lname, ssn);
        _weeklySalary = salary;
    }

    public double weeklySalary () {return _weeklySalary;}
    
    public void setWeeklySalary (double salary) {_weeklySalary = salary < 0.0 ? 0.0 : salary;}

    public double earnings() {return weeklySalary();}

    public String toString () {
        return String.format("salaried employee: %s\n%s: $%,.2f", 
            super.toString(), "weekly salary", weeklySalary());
    }
}

File: cse241-object-oriented-programming/studies/cpp/ch15-Polymorphism-and-Virtual-Functions/employee/salariedEmployee.cpp
#include <iostream>
using std::cout;
#include "salariedEmployee.h"

namespace GTUEmployee {
    SalariedEmployee::SalariedEmployee (const string & firstName, const string & lastName, const string ssn, double weeklySalary)
    : Employee(firstName, lastName, ssn) {
        setWeeklySalary(weeklySalary);
    }

    void SalariedEmployee::setWeeklySalary (double weeklySalary) {
        _weeklySalary = (weeklySalary < 0.0) ? 0.0 : weeklySalary;
    }
    
    double SalariedEmployee::getWeeklySalary () const {return _weeklySalary;}

    double SalariedEmployee::earnings () const {
        return _weeklySalary;
    }

    void SalariedEmployee::print() const {
        cout << "salaried employee: ";
        Employee::print(); // reuse abstract base-class print function
        cout << "\nweekly salary: " << getWeeklySalary();
    }
}